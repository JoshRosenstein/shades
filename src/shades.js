import css from './style-parser';
export * from './style-parser';
export default css;
