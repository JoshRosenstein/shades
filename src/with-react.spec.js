import React from 'react';
import { mount, render } from 'enzyme';
import shades, { Shades } from './with-react';

describe('Shades DOM', () => {
  const mountShades = (data) => mount(
    <Shades to={document.querySelector('head')}>
      {data}
    </Shades>
  );

  it('Renders without incident', () => {
    const Title = shades.h1({
      backgroundColor: {
        dark: 'blue',
        light: 'green',
        default: 'pink'
      },
      fontWeight: {
        dark: 600,
        light: 200
      },
      color: 'black'
    });

    const subject = mountShades(
      <Title dark>Hello</Title>
    );

    expect(subject).toMatchSnapshot();
  });
  it('Forwards valid DOM props', () => {
    const Linky = shades.a({
      backgroundColor: {
        dark: 'blue',
        light: 'green',
        default: 'pink'
      },
      fontWeight: {
        dark: 600,
        light: 200
      },
      color: 'black'
    });

    const subject = mountShades(
      <Linky href="hello.html" superDark="yeah">Hello</Linky>
    );

    const anchorItem = subject.find('a');;

    expect(anchorItem).toHaveProp('href', 'hello.html');
    expect(anchorItem).not.toHaveProp('superDark');
  });
});
