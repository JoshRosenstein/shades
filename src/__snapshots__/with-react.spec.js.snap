// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shades DOM Renders without incident 1`] = `
<Shades
  to={
    <head>
      <style
        data-shades="true"
      />
    </head>
  }
>
  <Component
    to={
      <head>
        <style
          data-shades="true"
        />
      </head>
    }
  >
    <pure(getContext(shades.h1))
      dark={true}
    >
      <getContext(shades.h1)
        dark={true}
      >
        <shades.h1
          dark={true}
          targetDom={
            <head>
              <style
                data-shades="true"
              />
            </head>
          }
        >
          <h1
            className="shades-1-0a5820ffa310da513618d2a4f400ba05be7c12cb "
          >
            Hello
          </h1>
        </shades.h1>
      </getContext(shades.h1)>
    </pure(getContext(shades.h1))>
  </Component>
</Shades>
`;
